Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: Rename UVM helper function radix_tree_empty to radix_tree_empty_nv
 Kernel 4.7-rc1 introduced its own radix_tree_empty function, so now Nvidia's
 internal helper function clashes. Rename by adding a _nv suffix.

--- a/nvidia-uvm/uvm_linux.h
+++ b/nvidia-uvm/uvm_linux.h
@@ -547,7 +547,7 @@
     INIT_RADIX_TREE(tree, GFP_NOWAIT);
 }
 
-static bool radix_tree_empty(struct radix_tree_root *tree)
+static bool radix_tree_empty_nv(struct radix_tree_root *tree)
 {
     void *dummy;
     return radix_tree_gang_lookup(tree, &dummy, 0, 1) == 0;
--- a/nvidia-uvm/uvm8_gpu.c
+++ b/nvidia-uvm/uvm8_gpu.c
@@ -525,7 +525,7 @@
                    gpu->id, uvm_gpu_retained_count(gpu));
 
     // All channels should have been removed before the retained count went to 0
-    UVM_ASSERT(radix_tree_empty(&gpu->instance_ptr_table));
+    UVM_ASSERT(radix_tree_empty_nv(&gpu->instance_ptr_table));
 
     // Remove the GPU from the table.
     uvm_spin_lock_irqsave(&g_uvm_global.gpu_table_lock);
